frame/dapps-staking/src/lib.rs

```diff
@@ -857,6 +877,34 @@ impl<T: Trait> Module<T> {
         total_payout
     }
 
+    fn propagate_eras_staking_points_total(
+        contract: &T::AccountId,
+        src_era: &EraIndex,
+        dst_era: &EraIndex,
+    ) {
+        if <ErasStakingPoints<T>>::contains_key(src_era, contract) {
+            let untreated_points = <ErasStakingPoints<T>>::get(src_era, contract);
+
+            <ErasStakingPoints<T>>::mutate(&dst_era, &contract, |points| {
+                (*points).total += untreated_points.total.clone();
+            });
+        }
+    }
+
+    fn is_payable(era: &EraIndex, nominator: &T::AccountId) -> bool {
```

Comment:
I'd like to see documentation on the newly added Nominator reward validity determination. Also, there are magic numbers(or methods) and these should be able to be Injected from Trait I think.